packages/block-editor/src/components/rich-text/test/index.perf-test.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import { measurePerformance } from 'reassure';
+import { fireEvent } from '@testing-library/react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { RichText } from '@wordpress/block-editor';
+
+describe( 'RichText Performance', () => {
+	it( 'measures performance', async () => {
+		const onChange = jest.fn();
+
+		const changeTextOfRichText = ( richText, text ) => {
+			fireEvent( richText, 'focus' );
+			fireEvent( richText, 'onChange', {
+				nativeEvent: {
+					eventCount: 1,
+					target: undefined,
+					text,
+				},
+			} );
```

Comment:
This fires a single event of `onChange`, which makes total sense for test preparation. Speaking at a higher level, while a user this event fires several times for each character. This raises two questions for me:

1. Does it make sense to mimic typing by triggering `onChange` for each character of a text set or is that fruitless? 
2. Is rendering and testing `RichText` alone a meaningful performance test or do we need a deeper editor tree? 

For the first question, it could be moot. If 1 `onChange` invocations equals 1 render, then 20 `onChange` invocations, in theory, equals 20 renders. If they are equal, is there value in mimicking typing via multiple events? 

For the second question, this likely comes down to _where performance issues originate_. Is the problem that `RichText` renders to frequently due to internal state? Is it due to props from parents? Is it due to context changes? It is difficult to know the most appropriate way to test performance without knowing the origin issue we want to avoid. Avoiding (and tracking) unnecessary re-renders is a good practice, but to what end? 

I share this not to say this test is incorrect, but to begin a discussion as to where/how performance tests provide meaningful value.

Reply:
Good questions. 

> Does it make sense to mimic typing by triggering onChange for each character of a text set or is that fruitless?
I don't think it would be a bad thing to measure microinteractions like this. They may produce useful metrics, and the only negative I can see is spending too much time writing Reassure tests. In the beginning of this project, I  envisioned that the Reassure test suite would need to expand to potentially every component in order to measure performance effectively. I'm re-evaulating that notion as we continue to write tests and explore testing organization. For the purposes of this PR, I wasn't intending for this to be a very robust test case, but I would very much welcome further discussion of how the structure of these tests could evolve into a suite robust test cases.

> Is rendering and testing RichText alone a meaningful performance test or do we need a deeper editor tree?
This was very much on my mind while writing the test as well -- I was trying to think of ways to avoid false positives or false reinforcement within the test structure itself. Having done a small amount of performance testing with Reassure, I think considering the depth of the editor tree is a very good topic to explore (or potentially discuss with the creators of the library) to see if the depth of nodes makes an impact on performance results, particularly as it relates to tracing exactly where re-renders originate.